src/SmartDetectorsApplianceShared/ChildProcess/ChildProcessManager.cs

```
@@ -324,6 +355,59 @@ public ITracer CreateTracerForChildProcess(string[] args)
+        {
+            string tempFolder = Path.Combine(Path.GetTempPath(), TempFolderName, Guid.NewGuid().ToString("N"));
+            if (!Directory.Exists(tempFolder))
+            {
+                Directory.CreateDirectory(tempFolder);
+            }
+
+            return tempFolder;
+        }
+
+        /// <summary>
+        /// Cleanup the temp folder
+        /// </summary>
+        /// <param name="tempFolderPath">The temp folder used for the child process (created by <see cref="CreateTempFolder"/>)</param>
+        private void CleanupTempFolder(string tempFolderPath)
+        {
+            DirectoryInfo tempFolder = new DirectoryInfo(tempFolderPath);
+
+            // Delete the specified temp folder
+            this.TryDeleteFolder(tempFolder);
+
+            // Delete the all folders in the parent of the temp folder that were created over 6 hours ago
+            // This handles scenarios of a previous crash that prevented the folder from being deleted
+            tempFolder.Parent?.GetDirectories()
+                .Where(subFolder => (DateTime.UtcNow - subFolder.CreationTimeUtc).TotalHours > 6)
+                .ForEach(subFolder => this.TryDeleteFolder(subFolder));
+        }
+
+        /// <summary>
+        /// Try to delete the specified folder - does not throw exception in case of failure
+        /// </summary>
+        /// <param name="folder">The folder to delete</param>
+        /// <returns>True if the folder was successfully deleted, false otherwise</returns>
+        private bool TryDeleteFolder(DirectoryInfo folder)
+        {
+            try
+            {
+                folder.Delete(true);
+                this.tracer.TraceWarning($"Successfully deleted temp folder {folder.FullName}");
+                return true;
+            }
+            catch (Exception e)
+            {
+                this.tracer.TraceWarning($"Could not delete temp folder {folder.FullName}, error: {e.Message}");
```

>this.tracer.TraceWarning($"Could not delete temp folder {folder.FullName}, error: {e.Message}"); [](start = 16, length = 96)

please report an exception so we will see it. this can cause issues if deletes fail a lot

---

Thanks, good point

---
In reply to: [186689392](https://github.com/liadtal/Smart-Alerts/pull/150#discussion_r186689392) [](ancestors = 186689392)